test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/MapTest.scala

```diff
@@ -1419,6 +1419,23 @@ trait MapTest {
       assertEquals("def", mp.get("nullable"))
     }
   }
+
+  @Test def additionToKeySet(): Unit = {
+    val chm = factory.empty[String, String]
+    val set = chm.keySet()
+
+    expectThrows(classOf[UnsupportedOperationException], set.add("ONE"))
+    expectThrows(classOf[UnsupportedOperationException], set.addAll(ju.Arrays.asList("ONE")))
+    expectThrows(classOf[UnsupportedOperationException], set.addAll(ju.Arrays.asList(null)))
+    expectThrows(classOf[UnsupportedOperationException], set.add(null))
+
+    if (factory.allowsNullCollectionsAdding) {
+      expectThrows(classOf[UnsupportedOperationException], set.addAll(null))
+    } else {
+      // NPE on JVM, JSE on JS
```

Comment:
If the JVM exposes different behavior when passing `null` collection, we should probably just drop this specific test (it's arguably a degenerate case).

@sjrd for a second opinion.

Reply:
Yes, this is full-blown Undefined Behavior, because the *collection* itself is null, not its elements. This test should be dropped (also the "then" branch, and therefore the `allowsNullCollectionsAdding` factory flag).

Reply:
Removed assertion.
But it seems that `coll.addAll(nulll)` should throw NPE according to JavaDoc.
So filed https://github.com/scala-js/scala-js/issues/4399

Reply:
This isn't addressed. The entire `if` and `allowsNullCollectionsAdding` should be removed.